contracts/RelayerHub.sol

```diff
@@ -8,88 +8,189 @@ import "./System.sol";
+            require(value.length == 32, "length of dues mismatch");
+            uint256 newDues = BytesToTypes.bytesToUint256(32, value);
+            require(newDues > 0 && newDues < requiredDeposit, "the dues out of range");
+            dues = newDues;
+        } else if (Memory.compareStrings(key, "addManager")) {
+
+            require(value.length == 20, "length of manager address mismatch");
+            address newManager = BytesToTypes.bytesToAddress(20, value);
+            addManagerByGov(newManager);
+
+        } else if (Memory.compareStrings(key, "removeManager")) {
+
+            require(value.length == 20, "length of manager address mismatch");
+            address payable managerAddress = payable(BytesToTypes.bytesToAddress(20, value));
+            removeManagerByGov(managerAddress);
+
+        } else {
+            require(false, "unknown param");
+        }
+        emit paramChange(key, value);
+    }
+
+    function removeManagerByGov(address payable managerToBeRemoved) internal {
+        removeManagerHelper(managerToBeRemoved);
+    }
+
+    function removeManager() external onlyRegisteredManager {
+        // here the manager removes himself
+        removeManagerHelper(payable(msg.sender));
+    }
+
+    function removeManagerHelper(address payable managerAddress) internal {
+        // check if the manager address already exists
+        require(relayManagersExistMap[managerAddress], "manager doesn't exist");
+
+        address relayerAddress = managersAndRelayers[managerAddress];
+
+        delete (relayManagersExistMap[managerAddress]);
+        delete (managersAndRelayers[managerAddress]);
+
+        manager memory a = managers[managerAddress];
+        managerAddress.transfer(a.deposit.sub(a.dues));
+        address payable systemPayable = payable(address(uint160(SYSTEM_REWARD_ADDR)));
+        systemPayable.transfer(a.dues);
+
+        delete (managers[managerAddress]);
+        delete (managersRegistered[managerAddress]);
+
+        // emit success event
+        emit removeManagerByGovEvent(managerAddress);
+        if (relayerAddress != address(0)) {
+            emit removeRelayerEvent(relayerAddress);
+        }
+    }
+
+    function addManagerByGov(address managerToBeAdded) internal {
+        require(!relayManagersExistMap[managerToBeAdded], "manager already exists");
+        require(!isContract(managerToBeAdded), "contract is not allowed to be a manager");
+
+        relayManagersExistMap[managerToBeAdded] = true;
+
+        emit addManagerByGovEvent(managerToBeAdded);
+    }
+
+    function registerManager() internal onlyNonRegisteredManager {
+        require(msg.value == requiredDeposit, "deposit value is not exactly the same");
+        managers[msg.sender] = manager(requiredDeposit, dues);
+        managersRegistered[msg.sender] = true;
+        emit registerManagerEvent(msg.sender);
+    }
+
+    function addRelayer(address relayerToBeAdded) public onlyRegisteredManager noProxy {
```

Comment:
Is `noProxy` necessary here? since you already ensure that the manager can not be a contract.